ggml: softmax op: update the n_task calculation

snadampal opened the pull request:
updated the n_task calculation to use max number of threads possible. This has improved the prompt eval performance by around 5% for DOT kernels and by around 10% for MMLA kernels on AWS Graviton3 (r7g.16xl instance).

fixes#https://github.com/ggerganov/llama.cpp/issues/5103

ggerganov commented on ggml.c:
```diff
@@ -16597,7 +16597,7 @@ static int ggml_get_n_tasks(struct ggml_tensor * node, int n_threads) {
         case GGML_OP_SOFT_MAX:
             {
-                n_tasks = MIN(MIN(4, n_threads), ggml_nrows(node->src[0]));
+                n_tasks = MIN(n_threads, MAX(ggml_nrows(node->src[0]) - 1, 1));
```
This way, for 4 rows and 4 threads we will select `n_tasks == 3`
Shouldn't this be:

```suggestion
                n_tasks = MIN(n_threads, MAX(ggml_nrows(node->src[0]), 1));
```

snadampal replied:
I thought we wanted the n_tasks to be less than the num rows, that's why I was using 1 less and capped to 1. If it can match the num rows, the following would work too, wouldn't it? will there be any scenario with zero number of rows?
`n_tasks = MIN(n_threads, ggml_nrows(node->src[0]));`

> Now, we have only one function: ggml_get_n_tasks(), so the original issue is no longer relevant and in theory it should work with any n_task smaller than the number of rows

cebtenzzre replied:
I assume he meant less than *or* equal to.

ggerganov replied:
Yes, sorry for the confusion.

`n_tasks = MIN(n_threads, ggml_nrows(node->src[0]));` is good

snadampal replied:
thanks, I have updated the PR.

snadampal committed:
ggml: softmax op: update the n_task calculation

updated the n_task calculation to use max number of
threads possible. This has improved the prompt eval
performance by around 5% for DOT kernels and by
around 10% for MMLA kernels on AWS Graviton3.
